### PR TITLE
Web5 test vectors setup & integration

### PR DESCRIPTION
**Pull request by amika-sq**
Removes the copied test vectors, and instead pulls them in from the [web5-spec](https://github.com/TBD54566975/web5-spec) repository, which is added to this repo as a submodule.

Also added the necessary Github actions to upload artifacts and trigger a SDK report runner build.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,3 +18,13 @@ jobs:
       run: swift build
     - name: Run tests
       run: swift test
+    - name: Hash test vector inputs
```
Comment (nitro-neal) [inline .github/workflows/ci.yml:21]: can remove this

**Comment by nitro-neal** [comment]
looks good so far!

**Comment by nitro-neal** [comment]
```import CustomDump
import XCTest

@testable import Web5

final class Web5TestVectorsDidJwk: XCTestCase {

    func test_resolve() throws {
        let testVector = try TestVector<String, DIDResolutionResult>(
            fileName: "resolve",
            subdirectory: "test-vectors/did_jwk"
        )

        testVector.run { vector in
            let expectation = XCTestExpectation(description: "async resolve")
            Task {
                let didURI = vector.input
                let result = await DIDJWK.resolve(didURI: didURI)
                XCTAssertNoDifference(result, vector.output)
                expectation.fulfill()
            }

            wait(for: [expectation], timeout: 1)
        }
    }

}
```

so I see the test name is `test_resolve`.

For it to just 'work out of the box' and be the same as everything else the test name should be `resolve`

If this is not possible or a massive anti pattern it is ok to keep it and we can change the runner to accept this

**Review by nitro-neal — APPROVED** [review]

**Comment by amika-sq** [comment]
> ```
> import XCTest
> 
> @testable import Web5
> 
> final class Web5TestVectorsDidJwk: XCTestCase {
> 
>     func test_resolve() throws {
>         let testVector = try TestVector<String, DIDResolutionResult>(
>             fileName: "resolve",
>             subdirectory: "test-vectors/did_jwk"
>         )
> 
>         testVector.run { vector in
>             let expectation = XCTestExpectation(description: "async resolve")
>             Task {
>                 let didURI = vector.input
>                 let result = await DIDJWK.resolve(didURI: didURI)
>                 XCTAssertNoDifference(result, vector.output)
>                 expectation.fulfill()
>             }
> 
>             wait(for: [expectation], timeout: 1)
>         }
>     }
> 
> }
> ```
> 
> so I see the test name is `test_resolve`.
> 
> For it to just 'work out of the box' and be the same as everything else the test name should be `resolve`
> 
> If this is not possible or a massive anti pattern it is ok to keep it and we can change the runner to accept this

Unfortunately, this won't be possible. In order for a function to be considered a test case, it must have the `test` prefix.

**Pull request merged**